query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -1287,6 +1292,95 @@ void testQueryWithAvgForTdigestColumn() {
         executionContext);
   }
 
+  @Test
+  public void testQueryWithIntArrayFilter() {
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        buildSimpleQueryWithFilter(
+            createIntNotInFilter("Span.metrics.duration_millis", List.of(123))),
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis NOT IN (123)",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  public void testQueryWithLongArrayFilter() {
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        buildSimpleQueryWithFilter(
+            createLongInFilter("Span.metrics.duration_millis", List.of(10L))),
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
```

Comment:
I am not able to recollect, why we didn't add that support prior?
Can you once try these queries on Pinot for validation? If you have already tried, we can ignore this.

Reply:
Verified the query

Reply:
Verified the query